src/mrinufft/trajectories/display.py

```diff
@@ -88,34 +107,51 @@ def display_2D_trajectory(
     ax : plt.Axes
         Axes of the figure.
     """
+    # Setup figure and ticks
     Nc, Ns = trajectory.shape[:2]
-    ax = setup_2D_ticks(size, subfigure)
-    trajectory = trajectory.reshape((Nc, -1, 2))
+    ax = _setup_2D_ticks(figsize, subfigure)
+
+    # Display every shot
     for i in range(Nc):
         ax.plot(
             trajectory[i, :, 0],
             trajectory[i, :, 1],
             color=COLOR_CYCLE[i % NB_COLORS],
             linewidth=LINEWIDTH,
         )
-    if one_shot:
+
+    # Display one shot in particular if requested
+    if one_shot or type(one_shot) == int:
```

Comment:
Using `isinstance` is beat practice, and would check for True:

```suggestion
    if one_shot is True or insistance(one_shot, int):
```

Reply:
It is actually one of the few failing cases of isinstance, because in Python booleans are a subclass of integers. So it would actually not work with isinstance